### PR TITLE
chore(style): sync submodule exclusion list between tidy and rustfmt

### PR DESCRIPTION
**Pull request by ismailarilik**
As asked in the FIXME comments

### CODE REVIEW

File: src/tools/tidy/src/walk.rs [hunk 1]
```diff
@@ -7,7 +7,6 @@ use ignore::DirEntry;
 
 /// The default directory filter.
 pub fn filter_dirs(path: &Path) -> bool {
-    // FIXME: sync submodule exclusion list with rustfmt.toml
```
Comment (jieyouxu) [inline src/tools/tidy/src/walk.rs:10]: Discussion: @onur-ozkan should tidy be reading from rustfmt.toml or we somehow make sure that there's a single source of truth for this? (not for this PR)

**Review by jieyouxu — APPROVED** [review]
Thanks, r=me after PR CI is green.

File: src/tools/tidy/src/walk.rs (hunk 1 above)
Comment (ismailarilik) [reply src/tools/tidy/src/walk.rs:10]: Skip lists for `rustfmt` and `tidy` seem not overlapping but if `tidy` needs to include the skip list of `rustfmt`, I wanna work on it.

Comment (ismailarilik) [reply src/tools/tidy/src/walk.rs:10]: image

My other concern is that highlighted rows in the screenshot above (from `rustfmt.toml`) are not submodules. Maybe some time ago, they were and added to that list. But now they are managed inside the `rust` repo and `rustfmt` doesn't format them. I've tried to run `x fmt` for them and found a lot of errors. Shouldn't they be formatted like other files in the `rust` repo as well?

Comment (jieyouxu) [reply src/tools/tidy/src/walk.rs:10]: Some of them are not submodules but subtrees and managed by `josh`, so I don't think this is easily sync'd.

**Comment by jieyouxu** [comment]
But anyway, if we do want to sync it should be a follow-up, this PR itself is fine AFAICT.

@bors r+ rollup

File: src/tools/tidy/src/walk.rs (hunk 1 above)
Comment (jieyouxu) [reply src/tools/tidy/src/walk.rs:10]: IIRC we don't want tidy to rely on a toml parser because it needs to build fast, so probably not sync'd easily.

Comment (ismailarilik) [reply src/tools/tidy/src/walk.rs:10]: So how about relying on `.gitmodules` file for both rustfmt and tidy?

**Pull request merged**